sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGeneratorTest.java

```diff
@@ -73,6 +73,18 @@ public void generateBinaryQueryWithArrayContainsUsesUpper() {
         Assert.assertEquals(result.getQueryText(), " WHERE ARRAY_CONTAINS(UPPER(r.lastName), UPPER(@lastName0)) ");
     }
 
+    @Test
+    public void generateBinaryQueryWithStringEqualsDoesNotUseUpper() {
+        Criteria nameStartsWith = Criteria.getInstance(CriteriaType.STRING_EQUALS, "firstName",
+                Collections.singletonList("TREVOR"),
+                Part.IgnoreCaseType.ALWAYS);
```

Comment:
what about NEVER should it call stringequals(left,r ight, false) or stringequals(UPPER(left), UPPER(right))?

Reply:
for Never, it will be stringequals(left,right) (not only for string equals, but startswith, endswith as well), will add a test case.